### DISCUSSION TITLE
Android runtime does not support some JDK method，Please replace!

### DISCUSSION THREAD

**Discussion by iwangxiaodong** [belaban/JGroups#694] [General]
org.jgroups.util.Util.methodNameToAttributeName(final String methodName)  and  attributeNameToMethodName(String attr_name)
Used android does not support JDK9 method:
	java.util.regex.Matcher.**appendTail(StringBuilder sb)**

Please change!

Thanks!

@belaban

**Comment by belaban** [comment]
To rephrase: Android does not support `Matcher.appendTail()`?

**Comment by belaban** [comment]
These 2 methods have been in JGroups for years. Why is this relevant *now*?

**Reply by iwangxiaodong** [reply]
Android only support https://developer.android.com/reference/java/util/regex/Matcher#appendTail(java.lang.StringBuffer)

**Comment by belaban** [comment]
https://issues.redhat.com/browse/JGRP-2670

**Reply by tristantarrant** [reply]
Also, with JGroups now requiring JDK 11, I'm not sure this matters.

**Reply by iwangxiaodong** [reply]
Android JDK vs Oracle JDK: Some methods are missing！
Android only support https://developer.android.com/reference/java/util/regex/Matcher#appendTail(java.lang.StringBuffer)
missing appendTail(StringBuilder sb)

**Comment by belaban** [comment]
I assume he's using JGroups 4.x?

**Comment by belaban** [comment]
Anyway, fixed and backported to 4.x

**Reply by iwangxiaodong** [reply]
Thanks for Fixed！

**Comment by iwangxiaodong** [comment]
@belaban Sorry:
  org.jgroups.util.Util.methodNameToAttributeName(final String methodName) and attributeNameToMethodName(String attr_name):
    **m.appendReplacement(sb,...); // android 14+ support!**

Android 14+：
  https://developer.android.com/reference/java/util/regex/Matcher#appendReplacement(java.lang.StringBuilder,%20java.lang.String)

My android app is Android 8.0, use jgroups report: NoSuchMethodException!

If there is no value to modify, I will use the jgroups source code directly，Don't use maven.

Thanks!

**Comment by belaban** [comment]
Do you have a simple fix? Otherwise I suggest modify the src code and build yourself

**Reply by iwangxiaodong** [reply]
I am from src code build，only modify Util.methodNameToAttributeName(...) and attributeNameToMethodName(...)：
StringBuffer sb=new StringBuffer(); // StringBuilder sb=new StringBuilder();

Used in Android 8.0 app is works!

I submit a PR? or you directly modify!
